infrastructure/ai/providers/openai_provider.py

```
@@ -71,29 +77,57 @@ async def generate(
-                except Exception as e:
-                    # 某些网关在路径错误时可能不抛严格的 404 而是抛出其他错误，如果消息含有明确路径错误也尝试降级
-                    if "404" in str(e) or "Not Found" in str(e) or "400" in str(e) or "Account invalid" in str(e) or "INVALID_ARGUMENT" in str(e):
-                        logger.info(f"Gateway returned error for Responses API ({base_url}), falling back: {str(e)}")
-                        self.__class__._fallback_to_chat_cache.add(base_url)
-                    else:
-                        raise
+                    if use_responses:
+                        try:
+                            return await self._generate_via_responses(prompt, config)
+                        except (openai.NotFoundError, openai.BadRequestError) as e:
+                            logger.info(f"Responses API unsupported for {base_url}, falling back to chat completions: {str(e)}")
```

Ruff (0.15.15)

[warning] Use explicit conversion flag

Replace with conversion flag

(RUF010)